### ISSUE TITLE
child_process#fork() takes much longer time for npm

### ISSUE THREAD

**Issue by dkwingsmt** [atom/electron#4265]
## Description
Our project is using ``child_process#fork()`` to run npm commands, but it turns out taking much longer time than normal. 
I haven't tested if this problem occurs when running other scripts, but as shown in the test below, running the same code in pure ``node`` behaves normally.

## Reproduction
1. In a new folder, clone an empty app: ``git clone https://github.com/atom/electron-quick-start``
1. ``cd`` into that directory and install npm locally: ``npm i npm``
1. Open this app: ``electron .``
1. In the console, run the following code
```
var npmpath=require('path').join(__dirname,'node_modules','.bin','npm');
var startTime=new Date().getTime();
var cp=require('child_process');
var child = cp.fork(npmpath,['dist-tag','ls','npm'],{stdio:'pipe',silent:true});
child.stdout.on('data', function(data) {
    console.log('stdout: ' + data);
});
child.stderr.on('data', function(data) {
    console.log('stdout: ' + data);
});
child.on('close', function(code) {
    console.log('closing code: ' + code);
    console.log('time: ' + (new Date().getTime()-startTime));
});
```
On my computer it takes about 36 seconds,
![image](https://cloud.githubusercontent.com/assets/1596656/12667586/35996b1e-c613-11e5-9b30-cbe14d4d52f9.png)

On the other hand, to test it in pure node,

1. Save the script above in the root directory of the same app as ``run.js``
1. ``node run.js``

On my computer it takes about 1.2 seconds, which is about the same as directly running this command in command line.
![image](https://cloud.githubusercontent.com/assets/1596656/12667603/7d7a64c4-c613-11e5-90be-7a2ca47d3648.png)

## Platform
+ node 5.1.1
+ Chrome 47.0.2526.110
+ Electron 0.36.5
+ OS X 10.11.3

**Comment by zcbenz** [comment]
Most other JavaScript files run fine, it seems that some timer in npm is preventing the process to exit since the close time is always around 36s. Also it is interesting npm v2 always takes 26s to close.

**Comment by zcbenz** [comment]
This has been fixed by #4927, which will be included in the next release.
